### DISCUSSION TITLE
Hosting Pocketbase on a VM for 'Google Cloud'

### DISCUSSION THREAD

**Discussion by hAhl2** [pocketbase/pocketbase#809] [Q&A]
Mulling over a Proof of Concept.  Essentially weighing the complexities of integrating two systems via a frontend,

- Firebase
  - Hosting (of the frontend app)
  - Functions (to interact with existing data in Google)
  - Firestore (minimal use for non-relational data)
  - Authentication 

- Pocketbase
  - A nice API to query relational data
  - Realtime API
  - Data access rules
  - Easy to deploy (https://pocketbase.io/docs/going-to-production/)

Essentially the front end user will be interacting with two separate domains (could be subdomains?).  
Envision some wonkiness surrounding users, e.g. duplicate user management and simultaneously authenticating in two systems, but the user list is well-defined and does not change much.

Wondering if there any other high-level concerns to anticipate.  For example cross-origin resource sharing (probably have to configure CORS headers)?

An alternative to the two system approach would be to utilize and integrate the go 'google api' client into pocketbase?

**Comment by hAhl2** [comment]
> Mulling over a Proof of Concept. Essentially weighing the complexities of integrating two systems via a frontend,
> 
> * Firebase
>   
>   * Hosting (of the frontend app)
>   * Functions (to interact with existing data in Google)
>   * Firestore (minimal use for non-relational data)
>   * Authentication
> * Pocketbase
>   
>   * A nice API to query relational data
>   * Realtime API
>   * Data access rules
>   * Easy to deploy (https://pocketbase.io/docs/going-to-production/)
> 
> Essentially the front end user will be interacting with two separate domains (could be subdomains?). Envision some wonkiness surrounding users, e.g. duplicate user management and simultaneously authenticating in two systems, but the user list is well-defined and does not change much.
> 
> Wondering if there any other high-level concerns to anticipate. For example cross-origin resource sharing (probably have to configure CORS headers)?
> 
> An alternative to the two system approach would be to utilize and integrate the go 'google api' client into pocketbase?

Seems to not be the ideal solution.

**Reply by pragmatech** [reply]
Isn't bringing google into the equation sort of counter to PB's inherently sovereign appeal? or am I missing something?
ps- Ironic or otherwise, I run PB via docker, nginx on a little e2-micro on GCP. Works just fine.

**Reply by hAhl2** [reply]
> Isn't bringing google into the equation sort of counter to PB's inherently sovereign appeal? or am I missing something? ps- Ironic or otherwise, I run PB via docker, nginx on a little e2-micro on GCP. Works just fine.

It is, and the OP goes against PB's inherently simplistic design.  The other detail from the OP is that there was not the predilection to continue along the simplistic design (i.e. extending the PB framework with implementing googleAPI functionality in go, rather then try write firebase functions in typescrpipt, to interact with existing data in google)...   Another thing is that OP can wait for something like https://github.com/benallfree/pockethost to become more feature-complete.

**Reply by benallfree** [reply]
Thanks for the mention @hAhl2. If you have any features in mind for pockethost, please feel welcome to initiate a discussion or suggestion over in the pockethost project :)

I'm not sure what OP had in mind, maybe there is an existing code base and backend. I suppose if there is some data stored in Firestore or the realtime database, using the Firebase client to continue accessing it could make sense. Or as OP mentioned, proxying that stuff through some custom PocketBase API endpoints. I'm not sure what would be best but it sounds like an uncomfortable marriage of tech stacks. Good luck OP!
